app/controllers/sessions_controller.rb

```diff
@@ -3,23 +3,32 @@ class SessionsController < ApplicationController
   before_action :require_user, only: [:destroy]
 
   def new
-    redirect_to home_path if logged_in?
+    if logged_in?
+      flash[:error] = "You already signed in."
+      redirect_to home_path
+    end
   end
 
   def create
-    user = User.find_by(email: params[:email])
-    if user && user.authenticate(params[:password])
-      session[:user_id] = user.id
+    if logged_in?
+      flash[:error] = "You already signed in."
       redirect_to home_path
     else
-      flash[:error] = "Something wrong."
-      render :new
+      user = User.find_by(email: params[:email])
+      if user && user.authenticate(params[:password])
+        flash[:success] = "Welcome back, #{user.full_name}!"
+        session[:user_id] = user.id
+        redirect_to home_path
+      else
+        flash[:error] = "Something wrong."
```

Comment:
You want `flash.now` when rendering response. Otherwise it will pop also on the next request.